### PR TITLE
test: adding a multi-envoy test

### PR DESCRIPTION
**Pull request by alyssawilk**
Functionally this
 - handles the multi-envoy signal handler crash
 - skips instantiating a runtime singleton (off by default, must stay off until https://github.com/envoyproxy/envoy/issues/19847 is closed)

Multi-envoy does not correctly support runtime flags or deprecation stats due to https://github.com/envoyproxy/envoy/issues/19847 being incomplete.  It can still handle proxy traffic client - L1 - L2 - upstream as shown in test.

Risk Level: low
Testing: yes 
Docs Changes: n/a
Release Notes: n/a
Part of https://github.com/envoyproxy/envoy-mobile/issues/2003

### CODE REVIEW

File: source/server/config_validation/server.h
Comment (RyanTheOptimist) [inline source/server/config_validation/server.h:94]: I would have expected this to delegate to InstanceImpl::runtime(). But does that not work because ValidationInstance does not extend InstanceImpl?

File: test/integration/multi_envoy_test.cc
Comment (RyanTheOptimist) [inline test/integration/multi_envoy_test.cc:49]: Nice!

Would it also make sense (perhaps as a follow up?) to have a test in which we create 2 distinct Envoy instances, both in front of the same (or different, I suppose) upstreams. Then we could verify that requests flow through both, close one, and verify that the other still works?

**Review by RyanTheOptimist — APPROVED** [review]
LGTM. I think it might be a good idea to also get a review from someone more familiar with Runtime in order to assess the implications of making it no longer a proper Singleton. But it looks good from my perspective.

File: test/integration/multi_envoy_test.cc
Comment (alyssawilk) [reply test/integration/multi_envoy_test.cc:49]: yeah I also plan on having tests where I tear down the L2 and verify the L1 500s correctly, or tear down L1 and make sure I can talk directly to L2 -> client, to ensure lifetime issues are really sorted and regression tested against.

File: test/integration/multi_envoy_test.cc
Comment (alyssawilk) [reply test/integration/multi_envoy_test.cc:10]: it probably does and clang tidy will catch me :-P  Will fix.

File: source/server/config_validation/server.h
Comment (alyssawilk) [reply source/server/config_validation/server.h:94]: exactly.  arguably I should do the same restart flag checks here, but I'lll get there in the PR I flip the runtime guard true by default.

**Review by alyssawilk — COMMENTED** [review]
yeah for sure this gets second pass review, even with minimal code.
The impact of not having a singleton runtime should be documented in the PR description, and should be moot once the other PRs I have in flight land, at which point there _is_ no runtime singleton access in the code any more and I can fix up the final tests and flip the flag true.

File: test/integration/multi_envoy_test.cc
Comment (snowp) [reply test/integration/multi_envoy_test.cc:50]: Having some tests that verify different shutdown ordering would be useful I think, just to make sure that they can exist independently of each other

**Review by snowp — COMMENTED** [review]
Seems reasonable to me

What is the behavior if the two instances are run with different options? e.g. if the log output path is different between the two

**Comment by alyssawilk** [comment]
re logging: unsure.  Added to my list of things on https://github.com/envoyproxy/envoy-mobile/issues/2003 to do in follow up PRs as I don't think I can get it done today :-)

**Comment by rojkov** [comment]
/wait
on CI

**Comment by rojkov** [comment]
/wait
on merge conflict

**Review by RyanTheOptimist — APPROVED** [review]

**Pull request merged**
